Sasquatch/Sasquatch/ViewControllers/MSAnalyticsViewController.swift

```diff
@@ -241,16 +246,34 @@ class MSAnalyticsViewController: UITableViewController, AppCenterProtocol {
   }
   
   func initLatencyPicker() {
-    let latencyPosition = Latency.allTimeValues.index(of: UserDefaults.standard.integer(forKey: kMSTransmissionIterval))
-    self.latencyPicker = MSEnumPicker<Latency>(
-    textField: latencyField,
-    allValues: Latency.allValues,
-    onChange: { index in
-      UserDefaults.standard.setValue(Latency.allTimeValues[index], forKey: kMSTransmissionIterval)
-    })
-    self.latency = Latency.allValues[latencyPosition ?? 0]
-    self.latencyField.delegate = self.latencyPicker
-    self.latencyField.text = self.latency.rawValue
-    self.latencyField.tintColor = UIColor.clear
+    transmissionIntervalField.inputAccessoryView = self.toolBarForKeyboard()
+    let interval = UserDefaults.standard.integer(forKey: kMSTransmissionIterval) != 0 ? UserDefaults.standard.integer(forKey: kMSTransmissionIterval) : 3
+    transmissionIntervalField.text = String(interval)
+    updateIntervalLabel(transmissionInterval: interval)
+  }
+  
+  func secondsToHoursMinutesSeconds(seconds : Int) -> (Int, Int, Int) {
+    return (seconds / 3600, (seconds % 3600) / 60, (seconds % 3600) % 60)
+  }
+  
+  func toolBarForKeyboard() -> UIToolbar {
+    let toolbar = UIToolbar()
+    toolbar.sizeToFit()
+    let flexibleSpace = UIBarButtonItem(barButtonSystemItem: .flexibleSpace, target: nil, action: nil)
+    let doneButton = UIBarButtonItem(title: "Done", style: .done, target: self, action: #selector(self.doneClicked))
+    toolbar.items = [flexibleSpace, doneButton]
+    return toolbar
+  }
+  
+  @objc func doneClicked() {
+    updateIntervalLabel(transmissionInterval: Int(transmissionIntervalField.text!))
+    UserDefaults.standard.setValue(Int(transmissionIntervalField.text!), forKey: kMSTransmissionIterval)
+    dismissKeyboard(self.transmissionIntervalField)
+  }
+  
+  func updateIntervalLabel(transmissionInterval: Int?) {
+    let (h, m, s) = secondsToHoursMinutesSeconds(seconds: transmissionInterval ?? 3)
+    let convertInterval = String(format: "%02d:%02d:%02d",h ,m ,s)
+    transmissionIntervalLabel.text = "sec " + "(" + convertInterval + ")"
```

Comment:
It's better to use escape operator and one pair of quotes.

Reply:
No longer relevant